src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/CmdLineClient.kt

```diff
@@ -27,6 +33,33 @@ fun getRequestData(zipBytes: ByteArray, rootFileName: String, taskId: String): B
             .toByteString()
 }
 
+fun directoryExistingCheck(directory: File) {
+    if (!directory.exists()) {
+        throw IOException("tasksDir directory(name is $directory) doesn't exists")
+    }
+
+    if (!directory.isDirectory) {
+        throw IOException("tasksDir directory(name is $directory) actually isn't a directory")
+    }
+}
+
+fun zipDirectory(directory: File): ByteArray {
+    directoryExistingCheck(directory)
+
+    val byteOutput = ByteArrayOutputStream()
+    val output = ZipOutputStream(byteOutput)
+    for (fileInDirectory in directory.listFiles()) {
```

Comment:
Rename to `file` ? It is clear from the context where this file comes from.

Reply:
Done